packages/typespec-ts/test/modularUnit/scenarios/samples/parameters/bodyOptionalCheck.md

```diff
@@ -54,6 +52,76 @@ Raw json files.
 }
 ```
 
+## Provide generated operation options
+
+Generated operation options.
+
+```ts models:withOptions
+import { OperationOptions } from "@azure-rest/core-client";
+
+/** Optional parameters. */
+export interface ReadOptionalParams extends OperationOptions {
+  optionalQuery?: string;
+  widget?: BodyParameter;
+}
+```
+
+## Provide generated operations to call rest-level methods
+
+## Operations
+
+Should generate operations correctly:
+
+```ts operations
+import { TestingContext as Client } from "./index.js";
+import {
+  StreamableMethod,
+  PathUncheckedResponse,
+  createRestError,
+  operationOptionsToRequestParameters
+} from "@azure-rest/core-client";
+
+export function _readSend(
+  context: Client,
+  name: string,
+  requiredQuery: string,
+  options: ReadOptionalParams = { requestOptions: {} }
+): StreamableMethod {
+  return context.path("/{name}", name).post({
+    ...operationOptionsToRequestParameters(options),
+    queryParameters: {
+      requiredQuery: requiredQuery,
+      optionalQuery: options?.optionalQuery
+    },
+    body: !options["widget"]
+      ? options["widget"]
+      : bodyParameterSerializer(options.widget)
```

Comment:
why not just keep it align to use options.widget?

Reply:
i let it kept align with options["widget"] but pls note there are two patterns in our code options.widget and options["widget"] and they are the same.

Reply:
Will create an issue to track this.